### PR TITLE
[libevent] crt linkage fix

### PR DESCRIPTION
**Pull request by nshishov**
Fix build error:

```
The following packages will be built and installed:
    libevent[core]:x64-windows-static-md-v142
Starting package 1/1: libevent:x64-windows-static-md-v142
Building package libevent[core]:x64-windows-static-md-v142...
-- Downloading https://github.com/libevent/libevent/archive/release-2.1.11-stable.tar.gz...
-- Extracting source D:/vcpkg/downloads/libevent-libevent-release-2.1.11-stable.tar.gz
-- Applying patch fix-file_path.patch
-- Using source at D:/vcpkg/buildtrees/libevent/src/.11-stable-ff3495c81a
-- Configuring x64-windows-static-md-v142
-- Building x64-windows-static-md-v142-dbg
-- Building x64-windows-static-md-v142-rel
-- Performing post-build validation
Expected Debug,Dynamic crt linkage, but the following libs had invalid crt linkage:

    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/debug/lib/event.lib: Debug,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/debug/lib/event_core.lib: Debug,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/debug/lib/event_extra.lib: Debug,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/debug/lib/event_openssl.lib: Debug,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Dynamic crt linkage, but the following libs had invalid crt linkage:

    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/lib/event.lib: Release,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/lib/event_core.lib: Release,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/lib/event_extra.lib: Release,Static
    D:/vcpkg/packages/libevent_x64-windows-static-md-v142/lib/event_openssl.lib: Release,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 2 error(s). Please correct the portfile:
    D:\vcpkg\ports\libevent\portfile.cmake
-- Performing post-build validation done
Error: Building package libevent:x64-windows-static-md-v142 failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libevent:x64-windows-static-md-v142
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.
```

### CODE REVIEW

**Comment by nshishov** [comment]
Linux build issue isn't related to my pull-request at all.

File: ports/libevent/fix-crt_linkage.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index bf333f3..5bc2a54 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -230,7 +230,7 @@ endif()
+ 
+ if (${MSVC})
+     set(msvc_static_runtime OFF)
+-    if ("${EVENT_LIBRARY_TYPE}" STREQUAL "STATIC")
++    if ("${VCPKG_CRT_LINKAGE}" STREQUAL "STATIC")
```
Comment (JackBoosY) [inline ports/libevent/fix-crt_linkage.patch]: `$ {VCPKG_CRT_LINKAGE}` cannot be used in `CMakeLists.txt`, so you need to import them using
```
vcpkg_configure_cmake(
   ...
   OPTIONS -DVCPKG_CRT_LINKAGE=${VCPKG_CRT_LINKAGE}
)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libevent/fix-crt_linkage.patch (hunk 1 above)
Comment (nshishov) [reply ports/libevent/fix-crt_linkage.patch]: Thanks, added.

**Comment by JackBoosY** [comment]
Seems arrow cannot find port `brotli`.

**Comment by JackBoosY** [comment]
I can't reproduce the arrow failure on my machine, this could be a fake error.

**Review by JackBoosY — APPROVED** [review]

**Comment by Rastaban** [comment]
I think I have seen arrow fail before due to conflicts with other ports, I will try to look into this in the CI system later this week.

**Pull request merged**